source/game/system/KPadController.cc

```diff
@@ -110,56 +111,53 @@ void RaceInputState::reset() {
     trickRaw = 0;
 }
 
+bool RaceInputState::accelerate() const {
+    return !!(buttons & 0x1);
+}
+
+bool RaceInputState::trickUp() const {
+    return !!(trick & 0x10);
+}
+
 KPadGhostButtonsStream::KPadGhostButtonsStream()
-    : buffer(nullptr), sequenceCount(-1), size(RKG_INPUT_DATA_SIZE), state(2) {}
+    : currentSequence(std::numeric_limits<u32>::max()), state(2) {}
```

Comment:
I'm leaning towards keeping the current reading system, but I don't want to flat-out dismiss this. Can you walk me through the changes more in-depth?

Reply:
This is a good change, I think this is a lot cleaner.